intake_esm/core.py

```diff
@@ -120,6 +120,29 @@ def search(self, **query):
         ret.df = self._get_subset(**query)
         return ret
 
+    def __getitem__(self, key):
+        path_column_name = self._col_data['assets']['column_name']
+        columns = self.df.columns.tolist()
+        columns.remove(path_column_name)
+
+        if isinstance(key, str) and '.' in key:
+            key_parts = key.split('.')
+        else:
+            key_parts = [key]
+        N = len(columns)
+        # Pad key list for missing values so that len(key) == len(columns)
+        key_parts += ['*'] * (N - len(key_parts))
+        query = dict(zip(columns, key_parts))
+        # Filter out the query for valid entries
+        filtered_query = {k: v for k, v in query.items() if v != '*'}
+        return self.search(**filtered_query)
```

Comment:
@andersy005 You might also check to see if the `len(self.search(**filtered_query)) == 1` and then return an `Entry` right here (i.e., in `__getitem__`).  I think that is appropriate to do that in `__getitem__` and leave the return value of `self.search(...)` unchanged.